src/weather_station/clima_sensor_us.rs

```diff
@@ -0,0 +1,149 @@
+    alarm_active: bool,
+}
+
+impl ClimaSensorUS{
+    
+    pub fn new(config: &mut Config) -> Self{
+        let device_name = config.get::<string>("climasensor/device");
+        if device_name != "/dev/null" {
+            let mut s = Self {
+                device: device_name,
+                baudrate: config.get::<i32>("climasensor/baudrate"),
+                parity: config.get::<char>("climasensor/parity"),
+                data_bits: config.get::<i32>("climasensor/databits"),
+                stop_bits: config.get::<i32>("climasensor/stopbits"),
+                slave_id: config.get::<i32>("climasensor/salveid"),
+                ctx: null,
+                alarm_active: false,
+            };
+            s.init();
+            s
+        }else{
+            Self {
+                device: device_name,
+                baudrate: 0,
+                parity: 'N',
+                data_bits: 0,
+                stop_bits: 0,
+                slave_id: 0,
+                ctx: null,
+                alarm_active: false,
+            }
+        }
+    }
+
+    fn init(&mut self){
+        self.ctx = Modbus::new_rtu(self.device, self.baudrate, self.parity, self.data_bits, self.stop_bits).expect("Error accured while creating new RTU Object");
+        self.ctx.set_slave(self.slave_id).unwrap_or_else(|_| panic!("Error accured while setting slave-id to '{}'", self.slave_id));
+        self.ctx.rtu_set_serial_mode(SerialMode::RtuRS485).expect("Error accured while setting serial mode to RS485");
+        self.ctx.rtu_set_rts(RequestToSendMode::RtuRtsUp).expect("Error accured while setting RTS ti RTS-UP");
+        self.ctx.rtu_set_custom_rts(RequestToSendMode::RtuRtsUp).expect("Error accured while setting custom RTS-function");
+
+        self.ctx.connect().expect("Error accured while connecting to Clima-Sensor");
+    }
+
+    /// These functions create a single number out of an vector.
+    /// The first entry in the vector are the most significant bytes and the second entry are the least significant bytes
+
+    fn conv_vec_to_value_s(vec: Vec<i16>) -> i32{
+        (vec[0]<<16 | vec[1]).into()
+    }
+
+    fn conv_vec_to_value_u(vec: Vec<u16>) -> u32{
+        (vec[0]<<16 | vec[1]).into()
+    }
+
+    
+
+    /// This function should be called periodically to check the temperature and wind speed.
+    /// if temp > 30°C and no wind, then a warning should be issued
+    /// if temp > 35°C a warning should be issued
+    /// if temp < 20° the warning should be removed
+    /// (Input Register - 0x04) temp-reg address 0x76C1; typ S32; real_result = response_temp/10
+    /// (Input Register - 0x04) wind-reg address 0x7533; typ U32; real_result = response_wind/10
+    /// 
+    /// The return value is bool on success, true if alarm is active and false is alarm is not active
+    /// If no ctx is configured the this function returns always false, so no warning is triggered
+
+    pub fn handle_temperature(&mut self) -> Result<bool,Error> {
```

Comment:
Is more consistent and also more correct (as it handles more than temperature).

```suggestion
    pub fn handle(&mut self) -> Result<bool,Error> {
```